### ISSUE TITLE
'Run All' not working - Windows

### ISSUE THREAD

**Issue by Schepy** [JunoLab/atom-julia-client#597]
I am running Atom 1.40.1 and Julia-client 0.9.4. and Julia 1.1.1.

There is no reporting error at all. Client buttons like Start Local Julia or Stop Julia work.
However, as stated Run All doesn't. It does nothing.

I had this problem with all 0.9.x versions. Version 0.8.9 doesn't have this problem. I had to revert to older version few times already. 0.8.9 doesn't seem to be compatible with latest Atom, so I have no choice but to submit this report.

I think it is conflict with default atom keybinding for Windows. (crtl-shift-enter is insert row up).
I tried to fix this by changing keybindings, but without success.

Toolbar button Julia->Run All doesn't work either. Path to Julia is correct (the fact that Start Local Julia works confirms it).

Kind regards

**Comment by aviatesk** [comment]
> I had this problem with all 0.9.x versions. Version 0.8.9 doesn't have this problem. 

`Run-All` command isn't there in `0.8.x` versions, thus how did you confirm this ?

**Comment by Schepy** [comment]
In 0.8.x it is called Run File. I assumed it is just renamed command due to the same "play" icon in toolbar.

I don't see "Run File" in 0.9.x and Run File is what I need.

**Comment by aviatesk** [comment]
Yeah, I renamed `Run-File` into `Run-All` when bumping to version 0.9, but they should behave same way.

Could you try if the steps below works ?
- Boot julia via command `Julia-Client: Start-Julia`
- Run `Julia-Client: Run-All` command

**Comment by Schepy** [comment]
I what to put these commands?
I tried with cmd, powershell and REPL. They are not recognized.

Julia can be started by opening C:\Users\[User]\AppData\Local\Julia-1.1.1\bin\julia.exe

**Comment by aviatesk** [comment]
I meant, to run command from [Atom's command pallet](https://flight-manual.atom.io/getting-started/sections/atom-basics/#command-palette) (or from the tool-bar).

**Comment by Schepy** [comment]
I found pallet. There is Start Julia command and it works.
There is no Run All command.

At home I still have 0.8.9 working. In 0.8.9 there is Run File command in pallet.

**Comment by aviatesk** [comment]
Okay, then maybe you've cluttered the setting of `Julia Syntax Scopes`.
I recommend you just leave it as is unless you understand Atom well.

**Comment by Schepy** [comment]
0.9.x doesn't work properly at both my home and work PC (both Windows).

I can leave it as is if it will be fixed soon (in week or two).
Otherwise, I have to find alternative.

**Comment by aviatesk** [comment]
You can just go Julia-Client's settings and find `Julia Syntax Scopes`, and then modify it so that it would be the default value.

**Comment by aviatesk** [comment]
Gonna closing this issue since the source of the problem seems to be found.
Feel free to comment on this if you still have issues even after you successfully changed the config setting.

**Comment by Schepy** [comment]
I am not enthusiastic about closing this...
Julia Syntax Scopes: source.julia, source.weave.md, source.weave.latex
It is already default.

Additional info:
I have installed Julia 1.2 (and uninstalled 1.1.1 and changed path to new Julia.exe).
Nothing changed. Problem still persists.

**Comment by aviatesk** [comment]
You can't find the other commands like `Julia-Client: Run-HOGE` ?
And please make sure you have installed the language-julia package, and Julia-Client version is the latest.

**Comment by Schepy** [comment]
Language Julia 0.19.1 and Julia-Client 0.9.4 are installed. In fact everything is up-to date. 
There are commands like Settings, Debug Info, Kill Julia, Start Julia, Open a REPL, Open a Console, Clear Linter etc.
There are total 30 commands found (I counted) in pallet with "Julia_Client:", but no Run-HOGE nor Run-All.

**Comment by aviatesk** [comment]
Can you tell the results of the command `Editor: Log Cursor Scope` ?

**Comment by aviatesk** [comment]
On a file where you want to run `Julia-Client: Run-All` and stuff.

**Comment by Schepy** [comment]
Scopes at Cursor

text.plain.null-grammar

**Comment by pfitzseb** [comment]
Do you have `language-julia` installed and enabled? If not then please do install it :)

**Comment by pfitzseb** [comment]
Also, please provide the output of `Debug Info` when reporting issues, that makes helping much easier.

**Comment by Schepy** [comment]
I would provide debug info if there were any.

I found the problem. My file didn't have any extension.
Version 0.8.x was happy to run files without extension. There was no warring about this change.

Maybe it wouldn't hurt not to block runs due to file extension.

**Comment by aviatesk** [comment]
> Maybe it wouldn't hurt not to block runs due to file extension.

It's not blocked by file extension, but actually by grammar. The thing is, Atom is an editor for general languages, thus it's would be appropriate only to register julia commands to julia files (think about the case when you mistakenly press tool-bar button when you open markdown files).

So I recommend you register julia grammar to your files when you use Juno. You can set grammar via command `Grammar Selector: Show`, or status-bar, when the file isn't recognised as julia files :)

**Comment by Schepy** [comment]
OK. It works now :)
I think 0.8.x run files even without correct grammar selection, but that is not a problem now that I know how to avoid this problem.

Thank you for support!

**Comment by r-johns** [comment]
I am using Julia Version 1.8.0-beta3 (2022-03-29).  The file that is in the active editor window has a .jl extension, and the first time I clicked the "Run All" button it executed the file.  Afterwards, it would not execute.  The Run All from the pallet seems to do nothing.  No error messages, no output.  I tried it on a Julia program I had in a Jupyter notebook with the same result.

I am new to Julia, Juno and Atom and thought I must be doing something wrong in this new environment.  I was surprised the next day when I started Atom and hit "Run All", the program worked perfectly and produced two plots.  After that, dead again.  Is there an older, reliable version that works consistently?
